### PR TITLE
New LeanIMT `import`/`export` methods

### LINKED ISSUE #155: Add `import`/`export` methods to the `LeanIMT` class
**Is your feature request related to a problem? Please describe.**

Optimizing the process by storing tree hashes could significantly reduce computation time, particularly for expansive trees. This approach eliminates the need for repeated recalculations, e.g. streamlining tree initialization during application startup.

**Describe the solution you'd like**

The `LeanIMT` class may have two additional methods: `export` and `import`. The first could simply return a stringified JSON of the tree nodes. The second could import that JSON and initialize the nodes without calculating any hashes. Remember that all the nodes in the tree are stored in the `_nodes[][]` array.

The import method must check that the tree is empty, to prevent devs from mistakenly overwriting existing tree nodes with imported nodes.

**Additional context**

This functionality could be useful for applications such as [Bandada](https://bandada.pse.dev) and [Zupass](https://zupass.org/), where trees may get particularly large and it is necessary to initialize a tree instance each time the apps start.

### PR DESCRIPTION
**Pull request by cedoor**
## Description

This PR implements two new methods in the LeanIMT class: `import` and `export`. You can find more information in the related issue.

## Related Issue(s)

Closes #155 

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/imt/tests/lean-imt.test.ts [hunk 1]
```diff
@@ -317,5 +317,58 @@ describe("Lean IMT", () => {
...
+            const tree = new LeanIMT(poseidon, leaves)
+
+            const exportedTree = tree.export()
+
+            expect(typeof exportedTree).toBe("string")
+            expect(JSON.parse(exportedTree)).toHaveLength(4)
+            expect(JSON.parse(exportedTree)[0]).toHaveLength(5)
+        })
+
+        it("Should not import a tree if it the exported tree is not defined", () => {
```
Comment (0xjei) [inline packages/imt/tests/lean-imt.test.ts:331]:
```suggestion
        it("Should not import an undefined tree", () => {
```

File: packages/imt/tests/lean-imt.test.ts [hunk 2]
```diff
@@ -317,5 +317,58 @@ describe("Lean IMT", () => {
...
+            const tree = new LeanIMT(poseidon, leaves)
+
+            const fun = () => tree.import(undefined as any)
+
+            expect(fun).toThrow("Parameter 'nodes' is not defined")
+        })
+
+        it("Should not import a tree if it the exported tree is not a string", () => {
```
Comment (0xjei) [inline packages/imt/tests/lean-imt.test.ts:339]:
```suggestion
        it("Should not import a non-string type tree", () => {
```

File: packages/imt/tests/lean-imt.test.ts [hunk 3]
```diff
@@ -317,5 +317,58 @@ describe("Lean IMT", () => {
...
+            const tree = new LeanIMT(poseidon, leaves)
+
+            const fun = () => tree.import(1 as any)
+
+            expect(fun).toThrow("Parameter 'nodes' is not a string")
+        })
+
+        it("Should not import a tree if it is not empty", () => {
```
Comment (0xjei) [inline packages/imt/tests/lean-imt.test.ts:347]:
```suggestion
        it("Should not import a tree inside a non-empty tree", () => {
```

**Review by 0xjei — APPROVED** [review]
Awesome. Just some nitpicks on test cases names but you can go ahead and merge. Great work!

**Comment by cedoor** [comment]
> Awesome. Just some nitpicks on test cases names but you can go ahead and merge. Great work!

Thanks 🙏🏽 Test descriptions follow the same format as the others. I'd suggest opening a new issue for changing them since we would need to update many other descriptions.

**Comment by 0xjei** [comment]
> > Awesome. Just some nitpicks on test cases names but you can go ahead and merge. Great work!
> 
> Thanks 🙏🏽 Test descriptions follow the same format as the others. I'd suggest opening a new issue for changing them since we would need to update many other descriptions.

Makes sense, tracked on <issue>

**Pull request merged**